specifyweb/frontend/js_src/lib/components/QueryBuilder/FieldFilter.tsx

```diff
@@ -396,6 +397,12 @@ function In({
   );
 }
 
+const showComparisonOperatorsForString = userPreferences.get(
```

Comment:
(optional) since this is accessed in global scope, front-end will try to access this value before user preferences are fetched from the back-end. that means:
- if this is the first time specify is opened, `showComparisonOperatorsForString` will equal to the default value , regardless of what is actually set in the preferences
- if this is not the first time specify is opened (or it was opened for a different user before), the preferences that were previously cached will be used
  - that means, if you switched user account, the `showComparisonOperatorsForString` will equal to the pref value for the previous user
  - if the pref was changed on a different machine (or default prefs app resource was changed), the `showComparisonOperatorsForString` would be stale until page is reloaded again

solution:
- wrap `queryFieldFilters` in `f.store()` and replace `showComparisonOperatorsForString` variable with `showComparisonOperatorsForString` function - this way, by the time `queryFieldFilters` is accessed for the first time (which should be after preferences are loaded - but double check that), the preferences would already be loaded